### PR TITLE
Fix GUI bugs

### PR DESCRIPTION
**Pull request by ImUrX**
* Fixed updates being announced when it was a older update
* Fixed updates including pre-releases
* Fixed reset tutorial not showing continue button one `curIndex` before
* Fixed F1 not working for opening docs page

### CODE REVIEW

File: gui/src/App.tsx [hunk 1]
```diff
@@ -142,11 +143,13 @@ export default function App() {
   const [updateFound, setUpdateFound] = useState('');
   useEffect(() => {
     async function fetchReleases() {
-      const releases: Release[] = await fetch(
+      const releases = await fetch(
         `https://api.github.com/repos/${GH_REPO}/releases`
-      ).then((res) => res.json());
+      )
+        .then((res) => res.json())
+        .then((json: Release[]) => json.filter((rl) => !rl.prerelease));
```
Comment (TheDevMinerTV) [inline gui/src/App.tsx:150]: It'd be smart if we were to actually validate the objects here, for example with `zod`.

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]

File: gui/src/App.tsx (hunk 1 above)
Comment (TheDevMinerTV) [inline gui/src/App.tsx:150]: It'd be smart if we were to actually validate the objects here, for example with `zod`.

File: gui/src/App.tsx [hunk 2]
```diff
@@ -235,8 +238,8 @@ export default function App() {
       }
     }
 
-    document.addEventListener('keypress', onKeyboard);
-    return () => document.removeEventListener('keypress', onKeyboard);
+    document.addEventListener('keyup', onKeyboard);
+    return () => document.removeEventListener('keyup', onKeyboard);
```
Comment (TheDevMinerTV) [inline gui/src/App.tsx:242]: Please just move this into a different PR...

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]

File: gui/src/App.tsx (hunk 1 above)
Comment (ImUrX) [reply gui/src/App.tsx]: validated :p

File: gui/src/App.tsx (hunk 2 above)
Comment (ImUrX) [reply gui/src/App.tsx:242]: *makes 4 branches*

File: gui/src/App.tsx [hunk 3]
```diff
@@ -142,11 +143,13 @@ export default function App() {
   const [updateFound, setUpdateFound] = useState('');
   useEffect(() => {
     async function fetchReleases() {
-      const releases: Release[] = await fetch(
+      const releases = await fetch(
         `https://api.github.com/repos/${GH_REPO}/releases`
-      ).then((res) => res.json());
+      )
+        .then((res) => res.json())
+        .then((json: any[]) => json.filter((rl) => rl?.prerelease === false));
```
Comment (TheDevMinerTV) [inline gui/src/App.tsx:150]: no.

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]

File: gui/src/App.tsx (hunk 3 above)
Comment (ImUrX) [reply gui/src/App.tsx:150]: its better than bringing a whole dependency for one type check

File: gui/src/App.tsx [hunk 4]
```diff
@@ -142,11 +143,17 @@ export default function App() {
...
 
-      if (__VERSION_TAG__ && releases[0].tag_name !== __VERSION_TAG__) {
+      if (
+        __VERSION_TAG__ &&
+        typeof releases[0].tag_name === 'string' &&
+        semver.gt(releases[0].tag_name, __VERSION_TAG__)
```
Comment (TheDevMinerTV) [inline gui/src/App.tsx:155]: very cursed, no thank you.

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]

**Review by Louka3000 — APPROVED** [review]
